src/default-request-executor.js

```diff
@@ -87,7 +87,7 @@ class DefaultOktaRequestExecutor extends RequestExecutor {
   }
 
   requestHasTimedOut(request) {
-    return request.startTime && ((new Date() - request.startTime) > this.requestTimeout);
+    return (this.requestTimeout > 0) && request.startTime && ((new Date() - request.startTime) > this.requestTimeout);
```

Comment:
The spec does not call out what to do with negative numbers - this code will behave with "zero and negative numbers result in no timeout" where the spec only calls out zero for that case.

We should either update the spec or update this code to be a `!== 0`. Or, catch negative numbers as an invalid option upfront and throw a configuration error.